components/focus-trap/focus-trap.js

```diff
@@ -0,0 +1,20 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { FocusTrapMixin } from './focus-trap-mixin.js';
+
```

Comment:
Wondering a bit how useful the mixin actually is... doesn't seem to be providing much value over the element.

Reply:
Yeah, I'd started with the mixin and then introduced the component after.  I could definitely just be a component.

Reply:
Dialog actually has some custom logic for determining the first focusable element when it's coming from the body - it places focus on the first focusable element after the abort button (but is also inside the trap because it is part of the cycle).  In that case, the mixin is useful because it allows the consumer to provide a custom `focusFirst` method.

Maybe I could still move the mixin code into the component though.  The `d2l-focus-trap` could dispatch an event, `d2l-focus-trap-enter`, and dialog could listen for it an place focus accordingly.  🤔

Reply:
Yeah or you could put like a `d2l-focus-trap-skip` CSS class on elements you want it to ignore when doing its calculations.